### DISCUSSION TITLE
Strange problem starting to load bash shell wsl2 Ubuntu/Windows 11

### DISCUSSION THREAD

**Discussion by Gustavocgsg** [JanDeDobbeleer/oh-my-posh#4329] [Troubleshoot]
Hello friends, I would like to expose a small problem that I am trying to identify when customizing Shell Bash on my WSL2.

I'm using Ubuntu 22.04, and Windows 11 22h2, as far as PowerShell 7.3.7 goes, everything went fine; However, when customizing Bash, I ran into some difficulties but in the end I managed to make the changes, following the Oh-my-posh documentation. Below I will provide a brief summary so that you can understand the context of everything.

For both installations/configurations, I chose to use a manual Aba, to be a clean installation, that is, my notebook has Windows 11 Pro 22h2 recently formatted and fully updated, for bash I had to install curl and unzip, as they are essential to be able to continue customizing the proposals in the documentation.

There is still no step to using the command provided by curl, which is the one below, as it is using a simple user login added and the sudo command at the beginning of the expressions, see below:

**curl:**
`sudo curl -s https://ohmyposh.dev/install.sh | sudo bash-s`

When pasted into bash it reads and installs it in Oh-my-posh in the following location:

**oh-my-posh installation location:**
`/usr/local/bin`

So far everything has gone well, he even informs that the themes are located in the following place:

**Location of Themes:**
`oh-my-posh init {shell_name} --config /root/themes/{theme_name}.omp.json`

**However, when checking the contents of the root folder I realize that the correct path is the one I modified below:**
`oh-my-posh init {shell_name} --config /root/.cache/oh-my-posh/themes/{theme_name}.omp.json`

**Suggestion:** By the way, it would be interesting to make the necessary modifications to the install.sh file, I downloaded it and realized that there are some sections that need attention to give the necessary permissions and display the correct information to the end user. I'm going to outline what I “understood” below and reference it with a ---> I just ask that you take a look and see if the points I raised make sense.

Image with reference to modification suggestions in Bash Shell in the file install.sh

[https://github.com/Gustavocgsg/my-images/blob/main/shell-bash-Modifications.png](https://github.com/Gustavocgsg/my-images/blob/main/shell-bash-Modifications.png)

![Sugentão Install.sh](https://github.com/Gustavocgsg/my-images/blob/main/shell-bash-Modifications.png?raw=true "Sugentão Install.sh")

Well, then I make a small adaptation to the code provided by the oh-my-posh documentation, looking as follows:

Note the location of the Themes:
`eval "$(oh-my-posh init bash --config /root/.cache/oh-my-posh/themes/cloud-native-azure.omp.json)"`

As requested in the documents, I open the bash settings with the following command:

Automatically save and load themes:
`sudo nano ~/.bashrc
`
At the end of the file I put this command line and save the information.

However, for this path above to work correctly, you need to give permissions to the directory, so I chose to use the following 
command:

Command to give permissions to the directory:
`sudo chmod 777 root -R` 

(this equates to full directory permission)

It's probably not the best permission option I should use, but I was a little desperate.

With this adjustment made, the command works, and the work seems to have been executed correctly, however, after loading the Windows Terminal again, and opening the Ubuntu Tab (bash), for some reason that I am not aware of, it displays a supposed error instruction as you can see in the image below:

[https://github.com/Gustavocgsg/my-images/blob/main/erro-first-line.png](https://github.com/Gustavocgsg/my-images/blob/main/erro-first-line.png)

![Erro de linha no Bash](https://github.com/Gustavocgsg/my-images/blob/main/erro-first-line.png?raw=true?raw=true "Erro de linha no Bash")

This, however, only occurs in the first line, so much so that after giving a 'clear', it starts again without displaying the error.
As there is no information, I don't know where or how I can find out what is happening, so I know that no error should be ignored, I come here humbly to ask for help to understand what is happening, and teach how it happens. solve. I am very grateful in advance for your help and patience.

**Ps:** After reading some posts, I realized that there is a command that can help identify what is happening, below is what it says here.

**Command used:**
`oh-my-posh debug`

Image with reference to errors presented in Bash Shell

[https://github.com/Gustavocgsg/my-images/blob/main/shell-error-bash-Tudo.png](https://github.com/Gustavocgsg/my-images/blob/main/shell-error-bash-Tudo.png)

![Debug de erro](https://github.com/Gustavocgsg/my-images/blob/main/shell-error-bash-Tudo.png?raw=true "Debug de erro")

**Comment by JanDeDobbeleer** [comment]
The start error is one that's on your machine prior to starting oh-my-posh. we use `PIPESTATUS` as well, and there could be a startup command that's not exiting 100% correct (we've seen this before on WSL, oh-my-posh is only the reporter here).

About the cache dir, it uses a mechanism to identify the cache location on the machine according to [this](https://github.com/JanDeDobbeleer/oh-my-posh/blob/main/src/platform/shell_unix.go#L101) logic. We want to see if we can create a folder in the folder you mention, but if we are unsuccessful (due to rights), it has a fallback to the user's HOME folder.

**Reply by Gustavocgsg** [reply]
Hello JanDeDobbeleer, First of all, I want to thank you for your time, the final I believe will go well.

Regarding the error at the beginning, could it be related to some dependency?
 
As I started with it, I didn't end up installing package managers like Chocolatey, or even programs like Git, Node, Docker and the like?

How can I identify this supposed poorly initializing command?

Is there any command I can give that tells me something about what is happening?

Thank you in advance for your attention

**Reply by JanDeDobbeleer** [reply]
@Gustavocgsg you can start bash using `bash -x` to enable traces. That could give a clearer indication of what's going on. Alternatively, use `set -e` as the first statement inside your profile to stop execution where it errors.

**Reply by Gustavocgsg** [reply]
@JanDeDobbeleer Hello everything is fine ?

I did what you asked me to do, I'm using Windows Terminal, initially I applied the `bash -x` command, it generated a very large return, below is what I had to copy in .txt, as doing a print would be unfeasible...:

[Bash shell return with bash -x command](https://github.com/Gustavocgsg/my-images/blob/main/gustavocgsg%40GUSTAVO-DELL%20.txt)

Then I applied the `set -e` command, which in turn returned a shorter command, and so I will choose to display the print below:

https://github.com/Gustavocgsg/my-images/blob/main/image-set_e.png

![https://github.com/Gustavocgsg/my-images/blob/main/image-set_e.png](https://github.com/Gustavocgsg/my-images/blob/main/image-set_e.png?raw=true "retorno o set -e")

**Reply by JanDeDobbeleer** [reply]
@Gustavocgsg you can see that bash executes A LOT of commands before even executing oh-my-posh. Would either remove those, or find the culprit there.

**Reply by Gustavocgsg** [reply]
hey @JanDeDobbeleer 

I was trying to identify where the error could come from... but unfortunately, due to lack of experience, I can't identify it... =/, if it's not too much to ask, could you please point it out?

Do you think it has anything to do with directory permissions in WSL2?

**Reply by JanDeDobbeleer** [reply]
@Gustavocgsg I have no idea. All I see is that there's a lot of logic being executed when the shell starts, which is probably specific WSL, but I don't have access to a machine that can run WSL2.

**Reply by Gustavocgsg** [reply]
I'm getting the impression that it has to do with permissions, as wsl tends to block things a bit due to being too careful, there was a time before this that I used this configuration and everything went well. Until, unintentionally, I had placed a command to give permissions to a directory, but... I replaced the sign that should be a + with a - result of that? Well, WSL2 simply broke down, I had to redo everything from absolute scratch, in this case it formatted the entire notebook, and I tried once again and now it came to this... this line, I wonder if this would have an impact on usage? Since it's like you said, he's a "retortor"... The error may not interfere with executions in the bash wsl2 shell.

**Reply by JanDeDobbeleer** [reply]
@Gustavocgsg if the error disappears from the prompt on enter, or a successful command, there's no issue at all. It's more a nuisance than a problem.

**Reply by Gustavocgsg** [reply]
@JanDeDobbeleer , This is good, from what I understand it only occurs on the first line if you clear it, and continue working on bash it works normally, I'll just do a few more experiments on permissions and see if there will be any changes, if I notice something I'll get back to you. this way ! so that others who have similar problems can be helped =D

**Reply by Gustavocgsg** [reply]
hey @JanDeDobbeleer , Please ask me a question??? If I choose to use the Homebrew package manager, and configure oh-my-posh following the documentation, is it possible that it will give the same line error result when loading?

Would the same thought apply if you choose to use SZH as your default shell?

**Reply by JanDeDobbeleer** [reply]
Yes, it has nothing to do with oh-my-posh on its own.
